src/documents/tests/test_file_handling.py

```diff
@@ -0,0 +1,616 @@
+import datetime
+import os
+import shutil
+from unittest import mock
+from uuid import uuid4
+from pathlib import Path
+from shutil import rmtree
+
+from dateutil import tz
+from django.test import TestCase, override_settings
+
+from django.utils.text import slugify
+from ..models import Tag, Document, Correspondent
+from django.conf import settings
+
+
+class TestDate(TestCase):
+    deletion_list = []
```

Comment:
I feel like the `deletion_list` might be a bit fragile, at least if a new test is added and one forgets to call `add_to_deletion_list`.

One alternative I can think of is to have a media-root that is global to the test-case, e.g.:

```python
from tempfile import TemporaryDirectory

class TestDate(TestCase):
    media_root = TemporaryDirectory(prefix="paperless-test-")

    def tearDown(self):
        self.media_root.cleanup()

    # ...
    @override_settings(MEDIA_ROOT=os.join(media_root.name, "file-renaming")
    @override_settings(...)
    def test_file_renaming(self):
        # ...
```

I think that this would not only get rid of the deletion-list, making it easier to understand the existing tests and also easier to write new ones, it also helps in the tests being cross-platform. (I'm not sure if they currently are anyway, but we don't have to make the situation worse if we can help it.)

Reply:
This was indeed too fragile and I fixed this now. I found a way to handle this in setUp and hence this cannot be forgotten anymore.